src/eval.cpp

```diff
@@ -733,36 +718,14 @@ void jitc_eval_impl(ThreadState *ts) {
             schedule_groups.size() == 1 ? "" : "s");
 
     scoped_set_context_maybe guard2(ts->context);
-    scheduled_tasks.clear();
 
     for (ScheduledGroup &group : schedule_groups) {
         jitc_assemble(ts, group);
 
-        scheduled_tasks.push_back(jitc_run(ts, group));
-
-        if (ts->backend == JitBackend::CUDA) {
-            jitc_free(kernel_params_global);
-            kernel_params_global = nullptr;
-        }
+        jitc_run(ts, group);
     }
 
-    if (ts->backend == JitBackend::LLVM) {
-        if (scheduled_tasks.size() == 1) {
-            task_release(jitc_task);
-            jitc_task = scheduled_tasks[0];
-        } else {
-            jitc_assert(!scheduled_tasks.empty(),
-                        "jit_eval(): no tasks generated!");
-
-            // Insert a barrier task
-            Task *new_task = task_submit_dep(nullptr, scheduled_tasks.data(),
-                                             (uint32_t) scheduled_tasks.size());
-            task_release(jitc_task);
-            for (Task *t : scheduled_tasks)
-                task_release(t);
-            jitc_task = new_task;
-        }
-    }
+    ts->barrier();
```

Comment:
Explain in a few words what this barrier achieves / why it's needed. E.g. ~"Ensure all kernels launched here finish before subsequent ones can start" (if I understood correctly).

Reply:
Suggest rephrasing: "The barrier ensures that subsequently launched kernel can't start to run until all kernels in the current launch have finished."